### PR TITLE
MM-67304: Deduplicate expired-token DMs to prevent Jira bot spam

### PR DESCRIPTION
**Pull request by nang2049**
#### Summary
Hovering Jira links with a disconnected or expired account spammed the Jira bot DM channel with the same warning message.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-67304

#### QA
1. As a test user, run `/jira connect` and complete OAuth.
2. Revoke the OAuth grant at https://id.atlassian.com/manage-profile/security/connected-apps → "Mattermost Jira Plugin" → Revoke access.

**Expected:** exactly **one** DM appears, regardless of how many links you hover or how fast.

## Change Impact: 🟡 Medium

**Reasoning:** The change adds a KV-store-based deduplication mechanism for expired-token notifications in an authentication-adjacent flow; it’s localized to the token-expiration handler, covered by new concurrent tests, but introduces atomic KV usage and TTL reliance that present moderate operational risk.

**Regression Risk:** Moderate — the code is defensive (falls back to existing behavior if KV writes fail) and tests were updated to cover concurrent deduplication, but the deduplication depends on KV availability and correct TTL semantics which could surface edge cases in production. The change is invoked from a small number of callers, limiting blast radius.

**QA Recommendation:** Moderate manual QA recommended. Verify the reported QA steps (revoke OAuth and hover many links to confirm a single DM). Also test KV unavailability/latency, concurrent expirations, and behavior during the cooldown window to ensure no unexpected suppression or duplicate notifications.

*Generated by CodeRabbitAI*

[Review Change Stack](https://app.coderabbit.ai/change-stack/mattermost/mattermost-plugin-jira/pull/1306)

### CODE REVIEW

File: server/utils.go [hunk 1]
```diff
@@ -89,6 +92,14 @@ func (p *Plugin) CreateBotDMtoMMUserID(mattermostUserID, format string, args ...
 }
 
 func (p *Plugin) disconnectUserDueToExpiredToken(mattermostUserID types.ID, instanceID types.ID) {
+	key := mattermostUserID.String() + ":" + instanceID.String()
+	if _, loaded := p.expiredTokenInFlight.LoadOrStore(key, struct{}{}); loaded {
+		return
+	}
+	defer time.AfterFunc(expiredTokenNotificationCooldown, func() {
+		p.expiredTokenInFlight.Delete(key)
+	})
```
Comment (avasconcelos114) [inline server/utils.go]: I'm wondering if we're better off handling this by using the KV store with a TTL expiry set?

At least I think we'll be able to add some extra resilience (against server restarts or in HA environments) for not much added complexity to trigger the same early-return pattern, what do you think?

**Review by avasconcelos114 — COMMENTED** [review]
Great improvement to the current flow here! 
Just left one suggestion up for discussion prior to approving

File: server/utils.go (hunk 1 above)
Comment (nang2049) [reply server/utils.go]: Good call :)) went with KV TTL!

**Review by avasconcelos114 — APPROVED** [review]
Awesome work! LGTM

**Review by ogi-m — APPROVED** [review]
LGTM!
